packages/gestalt/src/BannerOverlay/subcomponents.js

```diff
@@ -0,0 +1,148 @@
+      dangerouslyDisableOnNavigation: () => void,
+    }) => void,
+  },
+}): ReactNode {
+  const textRef = useRef<null | HTMLElement>(null);
+  const [ellipsisActive, setEllipsisActive] = useState(false);
+
+  // There’s two attributes for HTML elements which we can use to check if the text is truncated, offsetHeight and scrollHeight. scrollHeight is the total scrollable content height, and offsetHeight is the visible height on the screen. For an overflow view, the scrollHeight is larger than offsetHeight. We can deduce that if the scrollHeight is larger than the offsetHeight, then the element is truncated.
+  const isEllipsisActive = (element: HTMLElement) =>
+    element.offsetHeight < element.scrollHeight || element.offsetWidth < element.scrollWidth;
+
+  const checkEllipsisActive = useCallback(() => {
+    if (textRef.current && !ellipsisActive && isEllipsisActive(textRef?.current)) {
+      setEllipsisActive(true);
+    } else if (textRef.current && ellipsisActive && !isEllipsisActive(textRef?.current)) {
+      setEllipsisActive(false);
+    }
+  }, [ellipsisActive]);
+
+  useEffect(() => {
+    checkEllipsisActive();
+
+    if (typeof window !== 'undefined') window.addEventListener('resize', checkEllipsisActive);
+
+    return () => {
+      if (typeof window !== 'undefined') window?.removeEventListener('resize', checkEllipsisActive);
+    };
+  }, [checkEllipsisActive]);
+
+  const isTruncated = !textElement && text && ellipsisActive;
+  const isTruncatedWithHelperLink = isTruncated && helperLink;
+
+  return (
+    <Fragment>
+      {textElement ?? null}
+      {!textElement && text ? (
+        <Text
+          inline
+          align="start"
+          color={textColor}
+          lineClamp={2}
+          ref={textRef}
+          // Set title prop manually if text is truncated
+          title={isTruncated && typeof text === 'string' ? text : undefined}
+        >
+          {text}
+          {helperLink ? (
+            <Fragment>
+              {' '}
+              <Text inline color={textColor}>
+                <Link
+                  accessibilityLabel={helperLink.accessibilityLabel}
+                  href={helperLink.href}
+                  onClick={helperLink.onClick}
+                  target="blank"
+                  display="inlineBlock"
+                >
+                  {helperLink.text}
+                </Link>
+              </Text>
+            </Fragment>
+          ) : null}
+        </Text>
+      ) : null}
+      {isTruncatedWithHelperLink ? (
+        <Text color={textColor}>
+          <Link
+            accessibilityLabel={helperLink?.accessibilityLabel ?? ''}
+            href={helperLink?.href ?? ''}
+            display="inlineBlock"
+            onClick={helperLink?.onClick}
+            target="blank"
+          >
+            {helperLink?.text}
+          </Link>
+        </Text>
+      ) : null}
+    </Fragment>
+  );
+}
+
+export function BannerOverlayImageThumbnail({
```

Comment:
Could we reuse these subcomponents from Toast directly as well?

Maybe instead of under Toast, they can be under /shared

Reply:
DONE